File: C07_WHSMail/WHSMailVB/WHSMailHost/frmMain.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace WHSMailHost
	Public Partial Class frmMain
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub mnuExit_Click(ByVal sender As Object, ByVal e As EventArgs) _
			Handles mnuExit.Click
			' exit the application
			Me.Close()
		End Sub

		Private Sub frmMain_Resize(ByVal sender As Object, ByVal e As EventArgs) _
			Handles MyBase.Resize
			' hide the form
			Me.Hide()
		End Sub

		Private Sub frmMain_Load(ByVal sender As Object, ByVal e As EventArgs) _
			Handles MyBase.Load
			' start the WCF service
			MyServiceHost.StartService()
		End Sub

		Private Sub frmMain_FormClosing(ByVal sender As Object, _
										ByVal e As FormClosingEventArgs) _
			Handles MyBase.FormClosing
			' stop the WCF service
			MyServiceHost.StopService()
		End Sub
	End Class
End Namespace

File: C07_WHSMail/WHSMailCS/WHSMailHost/frmMain.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace WHSMailHost
{
	public partial class frmMain : Form
	{
		public frmMain()
		{
			InitializeComponent();
		}

		private void mnuExit_Click(object sender, EventArgs e)
		{
			// exit the application
			this.Close();
		}

		private void frmMain_Resize(object sender, EventArgs e)
		{
			// hide the form
			this.Hide();
		}

		private void frmMain_Load(object sender, EventArgs e)
		{
			// start the WCF service
			MyServiceHost.StartService();
		}

		private void frmMain_FormClosing(object sender, FormClosingEventArgs e)
		{
			// stop the WCF service
			MyServiceHost.StopService();
		}
	}
}